server/src/main/java/org/apache/cassandra/sidecar/routes/AbstractHandler.java

```diff
@@ -331,4 +332,16 @@ public static String extractHostAddressWithoutPort(String address)
         }
         return address;
     }
+
+    protected StorageOperations getStorageOperations(String host)
+    {
+        CassandraAdapterDelegate delegate = this.metadataFetcher.delegate(host);
+        StorageOperations storageOperations = delegate == null ? null : delegate.storageOperations();
+        if (storageOperations == null)
+        {
+            throw cassandraServiceUnavailable();
+        }
+
+        return storageOperations;
+    }
```

Comment:
we already have a method that does something similar in this class, we should just generalize that method. I suggest renaming `ifMetricsOpsAvailable` , and the code would look like this instead

```
    protected <P> void ifAvailableFromDelegate(RoutingContext context,
                                               String host,
                                               Function<CassandraAdapterDelegate, P> mapper,
                                               Consumer<P> ifAvailable)
    {
        CassandraAdapterDelegate delegate = metadataFetcher.delegate(host);
        if (delegate == null)
        {
            context.fail(cassandraServiceUnavailable());
            return;
        }
        P applied = mapper.apply(delegate);
        if (applied == null)
        {
            context.fail(cassandraServiceUnavailable());
            return;
        }
        ifAvailable.accept(applied);
    }
```

The existing callsite would change to this: 

```
ifAvailableFromDelegate(context, host, CassandraAdapterDelegate::metricsOperations, operations -> {
```